charts/warpstream-agent/templates/service.yaml

```diff
@@ -32,3 +33,47 @@ spec:
     {{- end }}
   selector:
     {{- include "warpstream-agent.selectorLabels" . | nindent 4 }}
+...
+{{- if eq (include "warpstream-agent.deploymentKind" .) "StatefulSet" }}
```

Comment:
can you also add a variable to enable/disable this? Not ever deployment needs per pod services. Let's keep it disabled by default to not change the default behavior.

Reply:
Done. Added a flag